helper_apps/MediaFileChunker/fileChunker.js

```diff
@@ -117,31 +98,90 @@ function convertYoutubeToMp3Stream(video) {
+        quality: 'highestaudio',
+    });
 
-const processYoutubeUrl = async (url) => {
-    const info = await ytdl.getInfo(url);
-    const audioFormat = ytdl.chooseFormat(info.formats, { quality: 'highestaudio' });
+    return new Promise((resolve, reject) => {
+        ffmpeg(stream)
+            .audioBitrate(128)
+            .save(filename)
+            .on('progress', p => {
+                readline.cursorTo(process.stdout, 0);
+                process.stdout.write(`${p.targetSize}kb downloaded`);
+            })
+            .on('error', (err) => {
+                console.log('an error happened: ' + err.message);
+                reject(err);
+            })
+            .on('end', () => {
+                console.log(`\ndone, thanks - ${(Date.now() - start) / 1000}s`);
+                resolve(filename);
+            });
+    });
+}
+
+const ytdlDownload = async (url, filename) => {
+    return new Promise((resolve, reject) => {
+        const video = ytdl(url, { quality: 'highestaudio' });
+        let lastLoggedTime = Date.now();
+
+        video.on('error', (error) => {
+            reject(error);
+        });
+
+        video.on('progress', (chunkLength, downloaded, total) => {
+            const currentTime = Date.now();
+            if (currentTime - lastLoggedTime >= 2000) { // Log every 2 seconds
+                const percent = downloaded / total;
+                console.log(`${(percent * 100).toFixed(2)}% downloaded ${url}`);
+                lastLoggedTime = currentTime;
+            }
+        });
+
+        video.pipe(fs.createWriteStream(filename))
+            .on('finish', () => {
+                resolve();
+            })
+            .on('error', (error) => {
+                reject(error);
+            });
+    });
+};
 
-    if (!audioFormat) {
-        throw new Error('No suitable audio format found');
+const processYoutubeUrl = async (url) => {
+    try {
+        const outputFileName = path.join(os.tmpdir(), `${uuidv4()}.mp3`);
+        await ytdlDownload(url, outputFileName);
+        return outputFileName;
+    } catch (e) {
+        console.log(e);
+        throw e;
     }
 
-    const stream = ytdl.downloadFromInfo(info, { format: audioFormat });
-    // const stream = ytdl(url, { filter: 'audioonly' })
+    // const info = await ytdl.getInfo(url);
+    // const audioFormat = ytdl.chooseFormat(info.formats, { quality: 'highestaudio' });
+
+    // if (!audioFormat) {
+    //     throw new Error('No suitable audio format found');
+    // }
+
+    // const stream = ytdl.downloadFromInfo(info, { format: audioFormat });
+    // // const stream = ytdl(url, { filter: 'audioonly' })
 
-    const mp3Stream = convertYoutubeToMp3Stream(stream);
-    const outputFileName = path.join(os.tmpdir(), `${uuidv4()}.mp3`);
-    await pipeStreamToFile(mp3Stream, outputFileName); // You can also pipe the stream to a file
-    return outputFileName;
+    // const mp3Stream = convertYoutubeToMp3Stream(stream);
```

Comment:
Shouldn't we trim all these commented lines?

Reply:
sure, cleaning